knox-token-generation-ui/token-generation/app/token-generation.service.ts

```diff
@@ -73,14 +75,31 @@ export class TokenGenService {
         });
     }
 
+    isTokenHashKeyPresent(): Promise<boolean> {
+        let headers = new HttpHeaders();
+        headers = this.addHeaders(headers);
+        return this.http.get(this.metadataInfoUrl, { headers: headers})
+            .toPromise()
+            .then(response => {
+                return response['generalProxyInfo']?.['enableTokenManagement'] === 'true';
+            })
+            .catch((err: HttpErrorResponse) => {
+                console.debug('TokenGenService --> isTokenHashKeyPresent() --> ' + this.metadataInfoUrl + '\n  error: ' + err.message);
+                if (err.status === 401) {
+                    window.location.assign(document.location.pathname);
+                } else {
+                    return this.handleError(err);
+                }
+            });
+    }
     getSessionInformation(): Promise<SessionInformation> {
         let headers = new HttpHeaders();
         headers = this.addHeaders(headers);
         return this.http.get(this.sessionUrl, { headers: headers})
             .toPromise()
             .then(response => response['sessioninfo'] as SessionInformation)
             .catch((err: HttpErrorResponse) => {
-                console.debug('TokenManagementService --> getSessionInformation() --> ' + this.sessionUrl + '\n  error: ' + err.message);
+                console.debug('TokenGenService --> getSessionInformation() --> ' + this.sessionUrl + '\n  error: ' + err.message);
```

Comment:
Same thing, do we need to display error messages to users on console? There could be a case made where it might help troubleshoot issues faster but displaying messages on console is not my favorite way to display errors :)